File: Java/Bryann Valderrama/findPair.java
public class Main{

    public static void main(String[] args) {
        int lista[] = {1, 3, 4, 5, 8, 10, 11};
        int x = 3;
        findPair(lista, x); 
    }

    static boolean findPair(int lista[], int x) {
        int tamanio = lista.length;
        int i = 0, j = 1;
        while (i < tamanio && j < tamanio) {
            if (i != j && lista[j] - lista[i] == x) {
                System.out.println("Par encontrado: " + "(" + lista[j] + "-" + lista[i] + ")"); //Encuentra la primera ocurrencia
                return true;
            } else if (lista[j] - lista[i] < x) {
                j++;
            } else {
                i++;
            }
        }
        System.out.println("No existe el par que logre ser igual a n al restar ");
        return false;
    }
}

File: PY/Luciano_Chavarria/Algoritmos_de_Busqueda/FindPair.py
def FindPair(arr,x):
    size = len(arr)
    i,j = 0,1
    while i<size and j<size:
        if i != j and arr[j] - arr[i] == x:
            print(f"par encontrado en: ({arr[i]} - {arr[j]})")
            return True
        elif arr[j]-arr[i]<x:
            j+=1
        else:
            i+=1
    print("No existe el par que cumpla la suma de n")
    return False

arr = [1,8,30,40,100]
x = int(60)
FindPair(arr,x)
